harness/determined/pytorch/_pytorch_context.py

```diff
@@ -266,11 +266,14 @@ def _filter_named_parameters(self, optimizer: torch.optim.Optimizer) -> List:
     def _init_device(self) -> None:
         self.n_gpus = len(self.env.container_gpus)
         if self.hvd_config.use:
-            check.gt(self.n_gpus, 0)
-            # We launch a horovod process per GPU. Each process
-            # needs to bind to a unique GPU.
-            self.device = torch.device(hvd.local_rank())
-            torch.cuda.set_device(self.device)
+            if self.n_gpus > 0:
+                check.gt(self.n_gpus, 0)
+                # We launch a horovod process per GPU. Each process
+                # needs to bind to a unique GPU.
+                self.device = torch.device(hvd.local_rank())
+                torch.cuda.set_device(self.device)
+            else:
+                self.device = torch.device("cpu")
         elif self.n_gpus > 0:
```

Comment:
Is this multi-GPU, non-horovod codepath still valid?

Reply:
yes, it runs on non-distributed single GPU path.

Reply:
Oh right, `1` is greater than `0` 🤦‍♂️